diff -up linux-5.18/drivers/net/vboxnetadp/Makefile-header.gmk.1005~ linux-5.18/drivers/net/vboxnetadp/Makefile-header.gmk
--- linux-5.18/drivers/net/vboxnetadp/Makefile-header.gmk.1005~	2022-07-22 20:22:06.467663348 +0200
+++ linux-5.18/drivers/net/vboxnetadp/Makefile-header.gmk	2022-07-22 20:24:26.997066146 +0200
@@ -50,7 +50,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
+ifeq ($(filter-out x86_64 znver1 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
diff -up linux-5.18/drivers/net/vboxnetflt/Makefile-header.gmk.1005~ linux-5.18/drivers/net/vboxnetflt/Makefile-header.gmk
--- linux-5.18/drivers/net/vboxnetflt/Makefile-header.gmk.1005~	2022-07-22 20:22:06.468663351 +0200
+++ linux-5.18/drivers/net/vboxnetflt/Makefile-header.gmk	2022-07-22 20:24:04.457997532 +0200
@@ -50,7 +50,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
+ifeq ($(filter-out x86_64 znver1 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
diff -up linux-5.18/drivers/virt/vboxdrv/Makefile-header.gmk.1005~ linux-5.18/drivers/virt/vboxdrv/Makefile-header.gmk
--- linux-5.18/drivers/virt/vboxdrv/Makefile-header.gmk.1005~	2022-07-22 20:22:06.469663354 +0200
+++ linux-5.18/drivers/virt/vboxdrv/Makefile-header.gmk	2022-07-22 20:24:46.104125392 +0200
@@ -50,7 +50,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
+ifeq ($(filter-out x86_64 znver1 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
